apps/estuary/docker-compose.yml

```diff
@@ -0,0 +1,57 @@
+---
+version: '3.4'
+services:
+  zookeeper:
+    image: wurstmeister/zookeeper
+    ports:
+    - "2181:2181"
+  kafka:
+    image: wurstmeister/kafka
+    depends_on:
+    - zookeeper
+    ports:
+    - "9092:9092"
+    environment:
+    - "KAFKA_ADVERTISED_LISTENERS=INSIDE://:9094,OUTSIDE://${HOST_IP:-localhost}:9092"
+    - "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT"
+    - "KAFKA_LISTENERS=INSIDE://:9094,OUTSIDE://:9092"
+    - "KAFKA_INTER_BROKER_LISTENER_NAME=INSIDE"
+    - "KAFKA_CREATE_TOPICS=test-topic:1:1"
+    - "KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181"
+    healthcheck:
+      test: ["CMD-SHELL", "cat /opt/kafka/logs/server.log | grep \"Previous Leader Epoch was: -1\""]
+      interval: 10s
+      timeout: 20s
+      retries: 3
+  metastore:
+    image: smartcitiesdata/metastore-testo:development
+    depends_on:
+    - postgres
+    ports:
+    - "9083:9083"
+    command: >
+      /bin/bash -c "
+        /opt/hive-metastore/bin/schematool -dbType postgres -validate || /opt/hive-metastore/bin/schematool -dbType postgres -initSchema;
+        /opt/hive-metastore/bin/start-metastore"
+  postgres:
+    logging:
+      driver: none
+    image: smartcitiesdata/postgres-testo:development
+    ports:
+    - "5432:5432"
+  minio:
+    image: smartcitiesdata/minio-testo:development
+    ports:
+    - "9000:9000"
+  presto:
+    depends_on:
+    - metastore
+    - minio
+    image: smartcitiesdata/presto-testo:development
+    ports:
+    - "8080:8080"
+    healthcheck:
+      test: ["CMD-SHELL", "curl -s http://localhost:8080/v1/info | grep -q '\"starting\":false'"]
+      interval: 10s
+      timeout: 30s
+      retries: 10 
```

Comment:
Can this configuration be done with the canned libraries that @jeffgrunewald built for Divo?  I know there's one for kafka, not sure about presto.

Reply:
Presto is a tricky one b/c of the configuration that’s required and the fact that it more or less requires config files on the pods’ file system. We’ve abstracted that out with the “testo” images by baking them specifically for our own test environment requirements. 

In order to feel comfortable publishing these to Hex as a publicly available “pre-baked” package we’d need to create the image with a templating engine built in for writing out those files at runtime based on envars or requiring a config value pointing to a file in the project directory for mounting in.

Reply:
Can we not use the presto-testo image here?

Reply:
That’s what they’re using but you still need to setup the port mapping, envs it does accept and start/stop it. Presto-testo just encapsulates the config files.

Reply:
Thanks @jeffgrunewald for clarification!

Reply:
No, it is not possible.